### ISSUE TITLE
features and classifier

### ISSUE THREAD

**Issue by G0rg0ne** [G0rg0ne/Fennec#1]
This task requires a top-n classification (multi-classes) because taking the top class is not sufficient due to the complex nature of the sounds (spectrums). You can have an environment where multiple sounds can be mixed (which is technically a convolution ) .

- This issue should aim at increasing the dimension of the features that we extract from the dataset (see if we can use other techniques).

- Add a multi-class top-n classifier to the model instead of a top-1.

**Comment by G0rg0ne** [comment]
![image](https://github.com/user-attachments/assets/fa04ff66-0a5d-441c-a208-60b0798efdc4)

The loss converges! now let's aim for performance :)

**Comment by G0rg0ne** [comment]
# Define the temperature parameter
temperature = 0.5  # Adjust this value (T < 1 makes it steeper, T > 1 makes it flatter)

# Apply the temperature-scaled sigmoid
all_outputs.append(torch.sigmoid(outputs / temperature).cpu()

**Comment by G0rg0ne** [comment]
```
 Post-Processing Techniques
If you cannot modify the model but still need to adjust the logits:

Min-Max Scaling: Scale the logits to fit into a range that better matches the sigmoid threshold:
python
Copier le code
scaled_logits = (logits - logits.min()) / (logits.max() - logits.min())
Log Transformation: Apply a logarithmic transformation if the logits span orders of magnitude.
```

**Comment by G0rg0ne** [comment]
performance update 
![image](https://github.com/user-attachments/assets/82d52452-1c0d-4d96-9a00-569f1b8c7592)

**Comment by G0rg0ne** [comment]
Future improvements: 
replace current data loader/feature extraction with [TorchAudio](https://pytorch.org/audio/main/transforms.html) capabilities to optimize memory throughput
